packages/vue-instantsearch/src/components/InstantSearch.js

```diff
@@ -111,6 +116,13 @@ export default createInstantSearchComponent({
     };
   },
   render: renderCompat(function (h) {
+    if (!this.indexName && !this.compositionID) {
+      // eslint-disable-next-line no-console
+      console.error(
+        'Missing either "index-name" or "compositionID" property on "ais-instant-search"'
+      );
+    }
```

Comment:
I couldn't find a way to make either `indexName` or `compositionID` required, so this is the next best thing I could come up with :/

Reply:
could you use the validator function and check in each if it or the other one is present in props? Otherwise I'd leave this error over to InstantSearch tbh

Reply:
what do you mean by "the validator function"?

Reply:
from my understanding, the standard vue validator function only allows to validate values, not properties presence vs. each other

Reply:
makes sense. In the end I'd leave out this console warning, as no index name on root has a use case (if there's an index widget child). I think people can see that nothing is happening and if they intended to use composition, they would know it needs to be passed.